torch/distributed/_tensor/_xla.py

```diff
@@ -0,0 +1,126 @@
+  for mesh_idx, spec in enumerate(placement_spec):
+    if spec.is_shard:
+      # mesh_idx to tensor_idx (spec.dim)
+      sharding_spec[spec.dim] = mesh_idx
+    elif spec.is_replicate:
+      # spec.dim is already set to None by default
+      continue
+    else:
+      raise ValueError(f"Unsupported placement type: {type(spec).__name__}")
+  return tuple(sharding_spec)
+
+
+def distribute_tensor(
+    tensor: torch.Tensor,
+    device_mesh: Optional[DeviceMesh] = None,
+    placements: Optional[Sequence[Placement]] = None,
+) -> XLAShardedTensor:
+    """
+    Distribute a torch.Tensor to the `device_mesh` according to the `placements`
+    specified. The rank of `device_mesh` and `placements` must be the same.
+
+    Args:
+        tensor (torch.Tensor): torch.Tensor to be distributed. Note that if you
+            want to shard a tensor on a dimension that is not evenly divisible by
+            the number of devices in that mesh dimension, we use `torch.chunk`
+            semantic to shard the tensor and scatter the shards.
+        device_mesh (:class:`DeviceMesh`, optional): DeviceMesh to distribute the
+            tensor, if not specified, must be called under a DeviceMesh context
+            manager, default: None
+        placements (List[:class:`Placement`], optional): the placements that
+            describes how to place the tensor on DeviceMesh, must have the same
+            number of elements as `device_mesh.ndim`. If not specified, we will
+            by default replicate the tensor across the `device_mesh` from the
+            first rank of each dimension of the `device_mesh`.
+
+    Returns:
+        A :class:`XLAShardedTensor` object
+    """
+    # get default device mesh if there's nothing specified
+    dt_mesh = device_mesh or mesh_resources.get_current_mesh()
+    assert dt_mesh.device_type == 'xla'
+
+    # convert to XLA device mesh
+    xla_mesh = convert_to_xla_mesh(dt_mesh)
+    assert xla_mesh.mesh_shape == dt_mesh.mesh.shape
+
+    # convert tensor to the corresponding device type if it's not in that device type
+    if not tensor.is_meta:
+        tensor = tensor.to(device_mesh.device_type)
+    # set default placements to replicated if not specified
+    if placements is None:
+        placements = [Replicate() for _ in range(device_mesh.ndim)]
+    assert len(placements) != device_mesh.ndim, f"`placements` must have the same length as `device_mesh.ndim`! "
+            f"Found placements length: {len(placements)}, and device_mesh.ndim: {device_mesh.ndim}."
+    # convert placements to xla partition spec
+    partition_spec = convert_to_xla_partition_spec(tensor, placements)
+    assert len(tensor.shape) == len(partition_spec), "`partition_spec` from `placements` must have the same length as `tensor.length`! "
+            f"Found tensor shape length: {len(tensor.shape)}, and partition_spec length: {len(partition_spec)}."
+
+    global_tensor = tensor
+    if isinstance(tensor, DTensor):
```

Comment:
`distribute_tensor` takes a tensor (global tensor) as input instead of DTensor, so we should error out here instead.

Reply:
Do you want to support the fully replicated case, or just block for all `DTensor` type. It's just that `DTensor` is also a `torch.Tensor`.